assets/js/base/components/cart-checkout/product-details/index.tsx

```diff
@@ -1,17 +1,20 @@
 /**
  * External dependencies
  */
-import PropTypes from 'prop-types';
 import { kebabCase } from 'lodash';
 import { decodeEntities } from '@wordpress/html-entities';
+import type { ProductResponseItemData } from '@woocommerce/type-defs/product-response';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 
+interface ProductDetailsProps {
+	details: ProductResponseItemData[];
```

Comment:
According to this component's code, the `name` property of each item is optional, but the `ProductResponseItemData` type has it as always existing. I didn't investigate which one is right, but I think it should be consistent?

Reply:
Hmm, I see the issue was there in the PropTypes as well. Looking at the code I think they were wrong. :thinking:

Reply:
I think the `name` property is not actually optional, the code just caters for it being an empty string. If you look here https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4098/files#diff-61ea5241262bf8f97af73e78e1d15467091815e69a744abb33b04dfa2e0ec098R40, allowing `name` to be undefined would result in trying to concatenate undefined with some other string. 

The `name` could be an empty string, that's where this ternary operator comes in:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4098/files#diff-61ea5241262bf8f97af73e78e1d15467091815e69a744abb33b04dfa2e0ec098R33

Reply:
Aaah, makes sense, thanks!